addons/mail/static/src/core/web/mail_composer_attachment_list.js

```diff
@@ -1,11 +1,27 @@
 import { registry } from "@web/core/registry";
+import { useService } from "@web/core/utils/hooks";
 import {
     many2ManyBinaryField,
     Many2ManyBinaryField
 } from "@web/views/fields/many2many_binary/many2many_binary_field";
 
 export class MailComposerAttachmentList extends Many2ManyBinaryField {
     static template = "mail.MailComposerAttachmentList";
+    /** @override */
+    setup() {
+        super.setup();
+        this.mailStore = useService("mail.store");
```

Comment:
Note that this might be a problem to not wrap the mail store in `useState()`, so `useState(useServive("mail.store"))`. This is a bit technical unfortunately, but OWL re-renders templates automatically when observed data is changed, and services that contain reactive data needs to be wrapped in `useState()`. `mail.store` is such a heavy reactive service. Not having it can lead to components that do not re-render when they should, which can be a big waste of time.

I'm wondering if we could improve services to flag them whether they are reactive, and have the `useService()` hook automatically wrap in a `useState`... This might help not necessarily have to be aware of this details and it just works. If we explicitly want to not have the `useState()` (sometimes for performance reasons), then we can have options on `useService()` to by-pass this automatic `useState`. I might suggest this improvement at some point in the future :)